spec.md

```diff
@@ -26,6 +27,61 @@ Enter CloudEvents, a specification for describing event data in a common way.
 CloudEvents seeks to ease event declaration and delivery across services,
 platforms and beyond.
 
+# Design Goals
+
+CloudEvents are typically used in a distributed system to allow for services to
+be loosely coupled during development, then may be configured with
+tight-coupling for specific applications or dynamically configured at runtime.
+
+The goal of the CloudEvents specification is to define interoperability of event
+systems that allow cloud providers, third-party tools and services to provide:
+- new event sources
+- new event consumers
+- observability, such as monitoring, debugging and visualization,
+- developer tools and libraries
+
+## Key properties of the system:
+- Source & action may be developed and deployed independently
+- Trigger configuration allows for actions to be invoked based on a condition
```

Comment:
I agree with the general goal of event sources and event handlers to be developed and deployed independently, but it's not clear to me how that goal can be specifically tied to this specification. You can easily sabotage this goal for yourself by expressing the event payload in a proprietary fashion that causes hard coupling between event source and handler, and there's little this spec can do to help you out of that. 

The core elements of the current specification do not define or mandate a "source"/"trigger"/"action" in the sense of resources that can be "developed and deployed". Those are very reasonable naming choices for concepts of a particular implementation that is using the specification, but so are "exchange"/"binding"/"queue" (RabbitMQ), "address"/"binding"/"queue" (ActiveMQ), "topic"/"selector"/"subscriber" (JMS), "topic"/"rule"/"subscriber" (Azure Service Bus), "topic"/"filter"/"subscriber" (Azure Event Grid), etc. 

* as an event publisher I want to make the event available in a way that it can be distributed to an interested audience and the audience can get at a particularly interesting subset with minimal own effort (i.e. delegate filtering to generic middleware)
* as an event consumer I want to have enough fidelity in the event metadata so that I can tell that generic middleware to pick out, and deliver a subset of events to me, and for there to be enough fidelity in the metadata for local dispatch to a handler.

I prefer for the spec not to pick favorites for what code concept might produce an event, what code concept might filter/dispatch an event, and what code concept might handle the event. We can certainly talk about concrete examples like sensors or cash registers and their respective contexts on one hand and about web hooks as event handlers on the other, but I prefer for us to stay neutral regarding the architecture model and nomenclature for how those events flow from A to B in a middleware usage scenario of this spec, precisely because there's already no shortage of those models.

(more context https://github.com/cloudevents/spec/issues/112 )